### ISSUE TITLE
Publish on Flathub

### ISSUE THREAD

**Issue by PanderMusubi** [tenacityteam/tenacity-legacy#25]
**Is your feature request related to a problem? Please describe.**
Publish on Flathub for distribution also via Pop!_OS.

**Describe the solution you'd like**
Publish on Flathub as the upstream of this fork does.

**Comment by CHJ85** [comment]
Let's at least wait til they have a proper name for this fork and rebrand it before creating all sorts of flatpak, appimages and snaps. Right?
As for now, the Ubuntu based repos are still using the older version of Audacity.

**Comment by dotjax** [comment]
> Let's at least wait til they have a proper name for this fork and rebrand it before creating all sorts of flatpak, appimages and snaps. Right?
> As for now, the Ubuntu based repos are still using the older version of Audacity.

I agree with this. Maybe keep the issue open and when the name is decided, someone can work on porting it to Flathub or other agnostic distribution systems. I've personally only managed .deb packages here or there, but I'd be willing to learn Flatpak once the name is decided.

**Comment by PanderMusubi** [comment]
Useful links

https://github.com/flathub/org.audacityteam.Audacity

https://www.flathub.org/apps/details/org.audacityteam.Audacity

https://github.com/flathub/flathub/wiki/App-Submission

**Comment by n0toose** [comment]
This will be useful for later down the line, but we really have to sort out things such as the name and get to writing code before we proceed with this.

**Comment by CHJ85** [comment]
@panos This is exactly why this fork should merge with Audacium. They already got the name and logo and everything figured out. No need for multiple forks.

**Comment by Semisol** [comment]
> @panos This is exactly why this fork should merge with Audacium. They already got the name and logo and everything figured out. No need for multiple forks.

One of the members here: Logo is a placeholder, but we are getting it figured out.

**Comment by TheEvilSkeleton** [comment]
I have some experience with Flatpak/Flathub so I can help.

**Comment by vchernin** [comment]
It’s probably best to create a new Flatpak for Tenacity, as it’s potentially confusing to try to replace Audacity. This seems to be what the Audacity Flatpak maintainers want anyhow https://github.com/flathub/org.audacityteam.Audacity/issues/72.

@TheEvilSkeleton do you think it makes sense to just reuse the Audacity Flatpak manifest for now, and submit that to flathub-beta until things have settled down more?

**Comment by TheEvilSkeleton** [comment]
As long as the name is final, then we can already create the manifest (obviously it will be based on Audacity's manifest). That being said, if this project ever gets deprecated, we can always archive the Flatpak package.

If we rush everything and at some point we change the name of the project, then we may have to use the old App ID but with a new name. A real life example is Element, formally known as Riot. The App ID is `im.riot.Riot` instead of `io.element.Element`, but the name of the app itself is Element.

IMO we should be focusing more on long term decisions, e.g. name of project, who has access to the repo, contributing guidelines, chat rooms, etc. and *then* focus on packaging. I only came by and said that I can help.

However, it's up to the maintainers to decide on what to focus. If they want to start packaging now, then sure, I'll help.

**Comment by vchernin** [comment]
Yeah, it’s unfortunate the Flathub name seemingly can’t be easily changed. From https://github.com/tenacityteam/tenacity/issues/94 it seems like Tenacity is final, and they did rename the repo. Chat rooms are being worked on in #7. Still things are moving very quickly, so it’s probably worth waiting at least a day or two before formally opening a PR at Flathub/Flathub.

**Comment by TheEvilSkeleton** [comment]
Yup. I'm gonna wait for a couple of days. When I have the time, I'll try to make a manifest and test it out.

**Comment by TheEvilSkeleton** [comment]
I started working on the manifest at https://github.com/TheEvilSkeleton/flatpaks/blob/com.github.tenacityteam.Tenacity/com.github.tenacityteam.Tenacity.yaml. I ran into an issue, which I assume is a Flatpak specific issue, so I opened the issue at https://github.com/flatpak/flatpak-builder-tools/issues/208.

**Comment by caughtquick** [comment]
@TheEvilSkeleton I would be happy to help maintain the flatpak.

**Comment by Semisol** [comment]
@TheEvilSkeleton may I have your Matrix tag/libera.chat nick? Planning on reaching out

**Comment by TheEvilSkeleton** [comment]
@Semisol `theevilskeleton` on Libera Chat.

**Comment by fossdd** [comment]
Any news?

**Comment by caughtquick** [comment]
> Any news?

Probably not going to Get it published till we make a proper release. I'm getting in contact with @TheEvilSkeleton to see if I can get there template. If not I'll just build my own to do it.

**Comment by TheEvilSkeleton** [comment]
I'm just waiting for https://github.com/tenacityteam/tenacity/pull/234 to get merged so I can start building the package without Conan. If you want to take over and continue the manifest now, you are free to fork my repo and continue.

**Comment by n0toose** [comment]
Also see: https://github.com/tenacityteam/tenacity/pull/228

**Comment by CHJ85** [comment]
@TheEvilSkeleton You might also wanna wait til they rebrand the app to Tenacity with the correct executable name as well as app name. That's like the first thing they should do. Because now if you install it as root and you got Audacity install, it'll overwrite the Audacity executable.

**Comment by TheEvilSkeleton** [comment]
@CHJ85 that's fortunately not a problem on Flatpak. Since all apps are in their own container, they're installed in separate directories, so you can have both Audacity and Tenacity flatpaks installed without any conflicts.

The executable's name can be rebranded later, even after we publish in Flathub. The main problem right now is actually building it and its dependencies from source.

**Comment by CHJ85** [comment]
@TheEvilSkeleton True. But they still haven't changed org.audacityteam.Audacity to something else. That is something they're gonna have to do first.

**Comment by TheEvilSkeleton** [comment]
@CHJ85 if you're talking about Audacity itself, then correct, they won't change the name anytime soon. However, this project is Tenacity and not Audacity, so it will be `org.tenacityaudio.Tenacity`. We haven't published it on Flathub yet, and won't do it until they fix the current problems with building, but after that, we will publish it with the `org.tenacityaudio.Tenacity` ID.

**Comment by CHJ85** [comment]
@TheEvilSkeleton No I'm talking about this project.
In the source code, they haven't changed org.audacityteam.Audacity, which they're gonna have to in order for the flatpak build to work. The files `AudacityApp.cpp, audacity.appdata.xml, AudacityCodeSigning.cmake, Build_Dist.sh, DMGSign.cmake, MacOSXBundleInfo.plist.in and signing.txt` include this string. This needs to be changed to whatever ID they decide to go with on Flathub.

**Comment by TheEvilSkeleton** [comment]
Ah, you're right, my bad.

Thank you for the info, I'm gonna wait for the whole rebranding to be done I guess.

**Comment by Be-ing** [comment]
Please wait until we have a first release until publishing to Flathub. There's still a bunch of work to do to unbork the build system.

**Comment by CHJ85** [comment]
@Be-ing Right. The rebranding can be done a bit more officiantly though. In order to fix this problem at hand, just run this command in the root directory of Tenacity master `find ./ -type f -exec sed -i -e 's/org.audacityteam.Audacity/org.tenacityaudio.Tenacity/g' {} \;`.
As for the name rebranding, run a similar command: `find ./ -type f -exec sed -i -e 's/Audacity/Tenacity/g' {} \;`.
This will obviously replace everything "Audacity" with "Tenacity" in all your files. So you're gonna have to rename the file names as well. This can be done easily by running a similar find and replace command: `find ./ -type f -exec rename 's/AUdacity/Tenacity/g' {} \;`.
As for the lowercase audacity files, I wouldn't change those, to avoid future conflicts with upstream Audacity.
Now when all that is taken care of, what's left is the executable and the logos and icons branding.
That should be about it.

**Comment by Be-ing** [comment]
The branding is not the only issue. We need to get the build system working as well and figure out how to migrate from Audacity user data.

**Comment by CHJ85** [comment]
@Be-ing Right. You have a long road ahead of you − so I'm glad you guys are being serious about this, because we really do need a good Audacity fork. So kudos!

**Comment by prateekmedia** [comment]
> I'm just waiting for #234 to get merged ...

It is now merged.

**Comment by n0toose** [comment]
Note: I received some feedback from a person who wanted a Flatpak wxgtk2 version of Tenacity. This should probably be taken into account.

**Comment by TheEvilSkeleton** [comment]
~~Do you need to manually disable Conan at build time or is it disabled by default?~~

Edit: disabled by default.

**Comment by TheEvilSkeleton** [comment]
Looks like I got it to work!

![image](https://user-images.githubusercontent.com/50847364/127778309-2bc2ac0d-7075-4598-b2a1-ecb508c23e60.png)

![image](https://user-images.githubusercontent.com/50847364/127778318-62e2dbd5-3569-4d1a-b698-ec9556a7b292.png)

**Comment by Semisol** [comment]
Nice I guess 🎉

**Comment by TheEvilSkeleton** [comment]
Hey maintainers, should I submit a draft MR on the Flathub repo or do I just wait? I know the software is in alpha, but I'd like some people to test the flatpak to check if everything works. Submitting an MR on the Flathub repo lets us create test packages so you can just install with `flatpak install`.

Here's the flatpak repo: https://github.com/TheEvilSkeleton/flatpaks/tree/org.tenacityaudio.Tenacity

**Comment by Be-ing** [comment]
Go ahead and make a merge request, but please mark it as a draft/work-in-progress so it doesn't get merged before we make a release.

**Comment by TheEvilSkeleton** [comment]
Don't worry, that was my intention.

**Comment by TheEvilSkeleton** [comment]
So Flathub has a beta branch, more info in their [repo](https://github.com/flathub/flathub/#using-the-flathub-repository).

I'd like to ask if we can publish experimental versions in the beta branch every couple of commits so we can have actual testers. What do you guys think?

**Comment by vchernin** [comment]
To add on since it wasn't explictly mentioned, anything exclusively in flathub-beta does not appear on [flathub.org](https://flathub.org). Users have to manually add the flathub-beta repo for apps to be visible in their software center/flatpak command line. So I personally think it's a good idea.
(if you noticed a weird edit I accidentally pressed my enter key while typing whoops)

**Comment by Be-ing** [comment]
Sounds good to me
